File: client_side-frontend/sit-pass/src/app/services/comment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { CommentDTO } from '../model/comment.model';
import { AuthService } from './auth.service';

@Injectable({
  providedIn: 'root'
})
export class CommentService {
  private apiUrl = 'http://localhost:8600/api/comments';

  constructor(private http: HttpClient, private authService: AuthService) {}

  createComment(comment: CommentDTO): Observable<CommentDTO> {
    const headers = this.authService.getAuthHeaders();
    return this.http.post<CommentDTO>(`${this.apiUrl}/createComment`, comment, { headers });
  }
}

File: server_side-backend/sis-pas/src/main/java/com/example/sispas/service/CommentService.java
package com.example.sispas.service;

import com.example.sispas.dto.CommentDTO;
import com.example.sispas.model.Comment;

public interface CommentService{
    CommentDTO createComment(CommentDTO commentDTO);
    CommentDTO toDTO(Comment comment);
    Comment toEntity(CommentDTO commentDTO);
}
